app/components/show/catalog_record_id_component.rb

```diff
@@ -1,17 +1,18 @@
 # frozen_string_literal: true
 
 module Show
-  class CatkeyComponent < ApplicationComponent
+  class CatalogRecordIdComponent < ApplicationComponent
     def initialize(change_set:, state_service:)
       @change_set = change_set
       @state_service = state_service
     end
 
-    def catkey
-      @change_set.catkeys.presence&.join(", ") || "None assigned"
+    def catalog_record_id
+      @change_set.catalog_record_ids.presence&.join(", ") || "None assigned"
     end
 
     delegate :allows_modification?, to: :@state_service
     delegate :id, to: :@change_set
+    delegate :manage_label, to: CatalogRecordId
```

Comment:
Another chunk of this PR is adding significant smarts to the `CatalogRecordId` class, attempting to concentrate knowledge of the separate implementations to as few spots as possible.